citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/CitationManager.java

```diff
@@ -494,4 +504,22 @@ public CitationPage getPrevAndNextCitation(IUser user, String groupId, String co
     public void deleteLocalGroupCitations(String groupId) {
         citationStore.deleteCitationByGroupId(groupId);
     }
+
+    @Override
+    public ICitation addCitationToReferences(ICitation citation, String referenceCitationKey, String reference) throws SelfCitationException {
+        Set<IReference> references = citation.getReferences();
+        if (references == null) {
+            references = new HashSet<>();
+        }
+        if (!citation.getKey().equals(referenceCitationKey)) {
+            IReference iReference = new Reference();
```

Comment:
why is this called `iReference`?

Reply:
IReference is interface and it's implementation is Reference for reference of the citation. It was already present in citesphere-model.